apps/oxlint/src/output_formatter/default.rs

```diff
@@ -132,8 +132,10 @@ mod test_implementation {
             let handler = GraphicalReportHandler::new_themed(GraphicalTheme::none());
             let mut output = String::new();
 
-            self.diagnostics.sort_by_key(|diagnostic| Info::new(diagnostic).filename);
-            self.diagnostics.sort_by_key(|diagnostic| Info::new(diagnostic).start.line);
+            self.diagnostics.sort_by_cached_key(|diagnostic| {
```

Comment:
also include the msg inside of the key? just for the case we're using for JS plugins where the loc, file name ect is the same, but the message is diff?

Reply:
We were writing at same time. OK, I'll try that and see if it changes the order anywhere else.

Reply:
I found I had to include `rule_id` too to get desired order in the JS plugins snapshots.